### PR TITLE
feat(core): post `/users`

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat(core): post `/users`

## Linear Issue Reference

* LOG-614

## Testing

Post with invalid body
postusers-invalid-body
Post with existed username
postusers-exist-username
Post with valid body
postusers-correct

### CODE REVIEW

File: packages/core/src/routes/admin-user.ts [hunk 1]
```diff
@@ -17,6 +20,46 @@ export default function adminUserRoutes<T extends AuthedRouter>(router: T) {
...
+        username: string().min(3),
+        password: string().min(6),
+        name: string().min(3),
+      }),
+    }),
+    async (ctx, next) => {
+      const { username, password, name } = ctx.guard.body;
+
+      assertThat(
+        !(await hasUser(username)),
```
Comment (simeng-li) [inline packages/core/src/routes/admin-user.ts:36]: nit: prefer avoid inline await 

`const hasUser = await hasUser(username);`

**Review by simeng-li — COMMENTED** [review]
overall LGTM

File: packages/core/src/routes/admin-user.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/routes/admin-user.ts:36]: inline await is commonly used in our code, do we need to discuss that?

Comment (xiaoyijun) [reply packages/core/src/routes/admin-user.ts:69]: LOG-1574

**Review by wangsijie — APPROVED** [review]

**Pull request merged**